src/components/tools/Partials/ChangeLocationStation/ChangeLocationStation.svelte

```diff
@@ -0,0 +1,155 @@
+
+  let currentLocation = location;
+  let currentBoundary = boundary;
+
+  let searchPlaceholder = isStationSelector
+    ? "Enter place name or address"
+    : boundary && boundary.metadata
+    ? updatePlaceholderText(boundary.metadata.placeholder)
+    : "";
+
+  function updatePlaceholderText(boundary) {
+    if (boundary && boundary.metadata) {
+      searchPlaceholder = `Enter ${boundary.metadata.placeholder}`;
+    }
+  }
+
+  async function handleSubmit() {
+    if (
+      (isStationSelector && currentLocation) ||
+      (!isStationSelector && currentBoundary && currentLocation)
+    ) {
+      open = false;
+      dispatch("change", {
+        ...(currentBoundary && { boundaryId: currentBoundary.id }),
+        location: currentLocation,
+      });
+    }
+  }
+
+  function handleSearchSelect({ detail }) {
+    currentLocation = detail;
+  }
+
+  function handleChangeBoundary({ detail }) {
+    currentBoundary = detail;
+    updatePlaceholderText(currentBoundary);
+  }
+
+  function handleMapClick({ detail }) {
+    if (detail && typeof detail === "object") {
+      currentLocation = detail;
+    }
+  }
+
+  function uploadBoundary({ detail }) {
+    currentBoundary = { id: "custom" };
+    currentLocation = detail.location;
+  }
+
+  function clearUpload() {
+    currentLocation = location;
+    if (currentBoundary.id === "custom") {
+      currentBoundary = boundary;
+    }
+  }
+
+  function cancel() {
+    currentLocation = location;
+    currentBoundary = boundary;
+    open = false;
+  }
+</script>
+
+<style>
+  .change-location {
+    position: relative;
+  }
+</style>
+
+<Modal
+  bind:open
+  on:click:button--secondary="{cancel}"
+  on:submit="{handleSubmit}"
+  on:open
+  on:close
+  preventCloseOnClickOutside
+  primaryButtonText="Confirm"
+  secondaryButtonText="Cancel"
+  modalHeading="{headingTitleText}"
+  shouldSubmitOnEnter="{false}"
+>
+  <div>
+    <p>{helpText}</p>
+    <!-- tools that use stations only have that type of boundary and no others -->
+    {#if !isStationSelector}
+      <SelectBoundary
+        on:upload="{uploadBoundary}"
+        on:clear="{clearUpload}"
+        on:change:boundary="{handleChangeBoundary}"
+        on:change:location="{handleSearchSelect}"
+        enableUpload="{enableUpload}"
+        currentBoundary="{currentBoundary}"
+        currentLocation="{currentLocation}"
+        boundaryList="{boundaryList}"
+        addStateBoundary="{addStateBoundary}"
+      />
+    {/if}
+    <div class="change-location">
+      <Search
+        on:select="{handleSearchSelect}"
+        isStationSelector="{isStationSelector}"
+        currentLayer="{isStationSelector ? stationsLayer : currentBoundary}"
+        searchPlaceholder="{searchPlaceholder}"
+        stationsLayerId="{isStationSelector ? stationsLayer.id : null}"
+      />
+
+      <LocationMap
+        on:click="{handleMapClick}"
+        {...{ currentLocation, currentBoundary, stationsLayer }}
+      />
+    </div>
```

Comment:
Might be able to place `<Search />` within `<LocationMap />`'s default slot? This would remove the wrapper `<div>` and CSS for `.change-location`.